docs/content/direct/user-guide.md

```diff
@@ -0,0 +1,75 @@
+# User Guide
+
+Status of this document: it is the barest of a start. Much more needs to be written.
+
+This document is for users of a release. Examples of using the latest stable release are in [the examples document](examples.md). This document adds information not conveyed in the examples.
+
+## Using a pre-existing cluster as the hosting cluster
+
+See [Using an existing hosting cluster](./hosting-cluster.md)
+
+## When everything is not on the same machine
+
+Thus far we can only say how to handle this when the hosting cluster is OpenShift. The problem is getting URLs that work from everywhere. OpenShift is a hosted product, your clusters have domain names that are resolvable from everywhere. In other words, if you use an OpenShift cluster as your hosting cluster then this problem is already solved.
+
+## Rule-based customization
+
+KubeStellar can distribute one workload object to multiple WECs, and it is common for users to need some customization to each WEC. By _rule based_ we mean that the customization is not expressed via one or more literal expressions but rather can refer to _properties_ of each WEC by property name. As KubeStellar distributes or transports a workload object from WDS to a WEC, the object can be transformed in a way that depends on those properties.
+
+At its current level of development, KubeStellar has a simple but limited way to specify rule-based customization, called "template expansion".
+
+### Template Expansion
+
+Template expansion is an optional feature that a user can request on an object-by-object basis. The way to request this feature on an object is to put the following annotation on the object.
+
+```yaml
+    control.kubestellar.io/expand-templates: "true"
+```
+
+The customization that template expansion does when distributing an object from a WDS to a WEC is applied independently to each leaf string of the object and is based on the "text/template" standard pacakge of Go. The string is parsed as a template and then replaced with the result of expanding the template. Errors from this process are reported in the status field of the Binding object involved.
+
+The data used when expanding the template are parameters describing the WEC. These come from two sources.
+
+1. The ConfigMap object, if any, that is in the namespace named "customization-properties" in the ITS and has the same name as the inventory object for the WEC. In particular, the ConfigMap string and binary data items whose name is valid as a [Go language identifier](https://go.dev/ref/spec#Identifiers) supply parameters.
+1. If not otherwise defined, there is a parameter whose name is "clusterName" and whose value is the name of the inventory item for the WEC.
```

Comment:
This is not clear to me. Only the "clusterName" parameter is supported?  I though every parameter that is not in the CM and does exist in the MC will be taken from there . 
It's also not clear where is it taken from, which object ?

Reply:
What do you mean by "every parameter that ... does exist in the MC"? Do you mean the labels and annotations?

Are you saying you are not sure what object is meant by the text "the inventory item for the WEC" ?  Would it help if that were expanded to "the inventory item for the WEC (i.e., the ManagedCluster object)"?

Reply:
1. Yes adding ManagedCluster object is perfect. Thanks
2. yes - I thought any "key" you use in the template , you first search for it in the CM and if not there you search over the labels and annotations in the ManagedCluster

Reply:
(A) oops, I see the old "parameter" language is still here. I changed it to "property" for consistency.

(B) I added a parenthetic remark to clarify. @nirrozenbaum will not like it because we expect generalization in the future. We will have to address this then.

Reply:
(C) I added a commit that adds inventory object labels and annotations as sources of WEC properties.

Reply:
@ezrasilvera what we agreed on when we discussed with the team was the start with config maps, without labels/annotations. even you agreed on this :)
I remember we even mentioned that hierarchy can be done using multiple config maps but as a starting point we start with a single configmap.
I understand the idea of hierarchy, but I have no doubt it complicates things. not only the implementation but also the users (not sure how intuitive is this direction for users). therefore the agreement was to start simple with only one configmap per WEC, and keep thinking about the hierarchy idea.

Reply:
@nirrozenbaum You are correct. I have no problem with that .   For some reason I was under the wrong assumption that we were adding CM and also reading from MC inventory objects. 
I agree it's probably also not good to count on MC that may be changed in the future when we introduce new inventory.  
I still think it would be good (maybe future PR ? ) if KS can create the initial CM with some default properties (e.g., ClusterName)

Reply:
@nirrozenbaum @MikeSpreitzer Sorry for the mess :(   That's what happen when someone tries to review without full awareness of the already resolved comments/discussions